spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/partitioner/CassandraPartitionedRDD.scala

```diff
@@ -26,11 +30,35 @@ class CassandraPartitionedRDD[T](prev: RDD[T])(implicit ct: ClassTag[T]) extends
     }
   }
 
+  /**
+   * This method currently uses thrift to determine the local endpoint and rpc endpoints
+   * from whatever endpoint the partition belongs to. This gives us a better chance of matching
+   * the bound interface of the Spark Executor. In addition we will add the HostName and
+   * HostAddress that we get for each of these endpoints to cover all of the logical choices.
+   */
   override def getPreferredLocations(split: Partition): Seq[String] = {
+    val connector = CassandraConnector(prev.sparkContext.getConf)
+
+    val rpcToLocalAddress = connector.withCassandraClientDo { client =>
+      val ring = client.describe_local_ring(keyspace)
+      for {tr <- ring
```

Comment:
code-style is slightly wrong:
http://docs.scala-lang.org/style/control-structures.html#curlybraces